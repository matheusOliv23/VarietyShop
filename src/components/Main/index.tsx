import React from "react"
import * as S from "./styles"

interface MainTypes {
  title: string
  description: string
}

export default function index({ title, description }: MainTypes) {
  return (
    <S.Wrapper>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <S.Illustration src="/astro2.jpg" alt="" />
    </S.Wrapper>
  )
}
