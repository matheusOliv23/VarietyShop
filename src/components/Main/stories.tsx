import React from "react"
import { ComponentStory, ComponentMeta } from "@storybook/react"

import Main from "./index"

export default {
  title: "Main",
  component: Main
} as ComponentMeta<typeof Main>

const Template: ComponentStory<typeof Main> = (args) => <Main {...args} />

export const Basic = Template.bind({})
Basic.args = {
  title: "Boilerplate",
  description: "Typescript, Reactjs, Nextjs, Styled Components"
}
