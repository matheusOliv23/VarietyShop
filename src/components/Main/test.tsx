import { render } from "@testing-library/react"
import Main from "./index"

describe("Main", () => {
  it("should render heading", () => {
    const { container } = render(
      <Main
        title="Boilerplate"
        description="Typescript, Reactjs, Nextjs, Styled Components"
      />
    )
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it("should render colors correctly", () => {
    const { container } = render(
      <Main
        title="Boilerplate"
        description="Typescript, Reactjs, Nextjs, Styled Components"
      />
    )
    expect(container.firstChild).toHaveStyle({ "background-color": "#06092b" })
  })
})
