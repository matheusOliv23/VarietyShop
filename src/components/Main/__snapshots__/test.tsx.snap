// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render heading 1`] = `
<div>
  <main
    class="sc-bczRLJ kIcxSX"
  >
    <h1
      class="sc-gsnTZi dUpkGV"
    >
      Boilerplate
    </h1>
    <h2
      class="sc-dkzDqf gOQRSK"
    >
      Typescript, Reactjs, Nextjs, Styled Components
    </h2>
    <img
      alt=""
      class="sc-hKMtZM bJYqql"
      src="/astro2.jpg"
    />
  </main>
</div>
`;
